app/components/auth/rc-auth-google/hooks/useRCAuth4Google.js

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from "react";
+
+import { useRouter } from 'next/router'
+import Cookies from "js-cookie";
+
+import RocketChatInstance from "../gapi";
+import { useGoogleLogin } from "../useGoogleLogin";
+
+export const useRCAuth4Google = () => {
+  const [user, setUser] = useState({});
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [userOrEmail, setUserOrEmail] = useState(null);
+  const [method, setMethod] = useState(undefined);
+
+  const { signIn, signOut } = useGoogleLogin(process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID);
+
+  const router = useRouter();
+
+  const RCInstance = new RocketChatInstance(
+    process.env.NEXT_PUBLIC_RC_URL,
+    process.env.NEXT_PUBLIC_RC_ROOM_ID
+  );
+
+  useEffect(() => {
+    const isStoredInSession = JSON.parse(sessionStorage.getItem("grc_user"));
+    if (isStoredInSession) {
+      setUser(isStoredInSession);
+    }
+  }, []);
+
+  const handleLogin = async (acsCode) => {
+    try {
+      const res = await RCInstance.googleSSOLogin(signIn, acsCode);
+      if (res?.me) {
+        sessionStorage.setItem("grc_user", JSON.stringify(res.me));
+        setUser(res.me);
+      }
+      if (res.error === "totp-required") {
+        setUserOrEmail(res.details.emailOrUsername);
+        setIsModalOpen(true);
+        if (res.details.availableMethods.length > 1) {
+          setMethod("totp");
+        } else {
+          setMethod(res.details.availableMethods[0]);
+        }
+      }
+    } catch (e) {
+      console.error("A error occurred while setting up user", e);
+    }
+  };
+
+  const handleLogout = async () => {
+    setUser({});
+    await RCInstance.logout(signOut);
+
+    sessionStorage.removeItem("grc_user");
+    router.reload();
+  };
+
+  const handleResend = async () => {
+    const res = await RCInstance.resend2FA(userOrEmail)
```

Comment:
Though I've added the resend function, while testing stuff out the current 2FA authentication doesn't work

While going through the documentation, I found that for 2fa authentication to a given route we need to add two headers...https://developer.rocket.chat/reference/api/rest-api/endpoints/other-important-endpoints/2fa#request-new-headers

> This is awesome work @Palanikannan1437 ! Please wire up the "Click to Chat" button top right for proper behavior.
> 
> It will demonstrate to all users how to use the auxiliary data obtained from Auth. (which is a major feature of the auth component)

Hey @Sing-Li  I've added the requested changes to the `Click to Chat` button, now

1. If user is logged in, the click to chat button opens up a new R.C instance window with the user already logged into rocket chat
2. If user is not logged in, the button redirects to the home page of the given R.C instance

Current Issues I'm facing,
1. Once say the user logs in to Rocket chat via google auth, I use the `authToken` provided by Rocket.Chat app instance to use it as a resume token...but after using the resume token once and say if the user logs out of the Rocket.Chat app instance, the current resume token doesn't work anymore....*how do I revalidate the resumetoken/accessToken?*....the only way I could think of is calling the api/v1/login route again for a new authToken

Reply:
Strange, the current implementation which you have added in this PR works perfectly fine.
Here,
<img width="173" alt="image" src="https://user-images.githubusercontent.com/61188295/193411452-6b85fe3f-2ca7-41a7-aa44-9f1b84bfe61a.png">

I am not sure why this error occurred for you. Ok, let's continue the conversation on Rocket Chat DMs.

Thank you!

Reply:
@Palanikannan1437 hey, can you also add 2fa into [EmbeddedChat](https://github.com/RocketChat/EmbeddedChat)? It will be quite similar process but it will really help the users to extend the Google SSO model there as is now.

Reply:
> @Palanikannan1437 hey, can you also add 2fa into [EmbeddedChat](https://github.com/RocketChat/EmbeddedChat)? It will be quite similar process but it will really help the users to extend the Google SSO model there as is now.

Ohh yeah sure! I'll try to do that asap!

Reply:
> > @Palanikannan1437 hey, can you also add 2fa into [EmbeddedChat](https://github.com/RocketChat/EmbeddedChat)? It will be quite similar process but it will really help the users to extend the Google SSO model there as is now.
> 
> Ohh yeah sure! I'll try to do that asap!

Thanks!!!